[TextFields] Make Kitchen Sink better for tests.

romoore opened the pull request:
Adding "validation" to make it easier to test error states in the
Kitchen Sink example.

Preparation for #7157

Commit:
[TextFields] Make Kitchen Sink better for tests.

Adding "validation" to make it easier to test error states in the
Kitchen Sink example.

Preparation for #7157

material-automation commented:
bazel detected changes to the following targets:

```
//components/TextFields:SwiftExamples
```

<!-- identifier: affected-targets -->

codeman7 commented on components/TextFields/examples/TextFieldKitchenSinkExample.swift:
```diff
@@ -690,6 +690,35 @@ extension TextFieldKitchenSinkSwiftExample: UITextFieldDelegate {
   }
+
+  func textFieldShouldClear(_ textField: UITextField) -> Bool {
+    for controller in self.allTextFieldControllers {
```
_Tip:_ Remove `self`

```diff
- self.allTextFieldControllers
+ allTextFieldControllers
```

romoore replied:
Done.

Commit:
Make A11Y value different from displayed.

This makes it easier to ensure we're using the a11y value and not the
displayed value when presenting information to VoiceOver.

codeman7 commented on components/TextFields/examples/TextFieldKitchenSinkExample.swift:
```diff
@@ -690,6 +690,35 @@ extension TextFieldKitchenSinkSwiftExample: UITextFieldDelegate {
+                 shouldChangeCharactersIn range: NSRange,
+                 replacementString string: String) -> Bool {
+    for controller in self.allTextFieldControllers {
+      if (textField == controller.textInput) {
```
_Tip:_ Remove the braces here and throughout
```diff
- (textField == controller.textInput)
+ textfield == controller.textInput
```

romoore replied:
Thanks. I really dislike Swift...

romoore replied:
Done.

Commit:
Swiftifying the code

Commit:
Using accessible functions
